framework/include/xfem/ElementFragmentAlgorithm.h

```diff
@@ -0,0 +1,92 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#ifndef ELEMENTFRAGMENTALGORITHM_H
+#define ELEMENTFRAGMENTALGORITHM_H
+
+#include <vector>
+#include <map>
+#include <set>
+
+#include "EFANode.h"
+#include "EFAElement.h"
+
+class ElementFragmentAlgorithm
+{
+public:
+
+  /**
+   * Constructor
+   **/
```

Comment:
This type of comment does not add much value. Comments on all the undocumented members below would be more useful.